shell.nix

```diff
@@ -47,6 +47,9 @@ mkShell {
     utillinux
     gnuplot
     libunwind
+    autoconf
+    automake
+    yasm
```

Comment:
not related, but at some point it'd be nice to make one shell nix depend on the other

Reply:
we actually need only one, "ci.nix". The default one "shell.nix" is for using system's Rust, which is not a good idea is general.  So I want to remove shell.nix completely and make it alias to ci.nix.
As for ability to build with the system's Rust, I'm planning to do it with the shell arg we already have: "--arg norust true".